src/monitor.ts

```
@@ -0,0 +1,357 @@
+   * the monitor will not try to restart the node.
+   */
+  reviveCmd?: string;
+
+  /**
+   * The arguments that should be passed to the cmd above, seperated by space.
+   */
+  reviveArgs?: string;
+};
+
+
+/**
+ * Takes any object and checks if it conforms to the config interface.
+ */
+function validateMonitoredNodeConfig(data: any) : IMonitoredNodeConfig | undefined {
+  if (!data.hasOwnProperty('address')) {
+    return undefined;
+  }
+
+  if (typeof(data.address) !== 'string') {
+    return undefined;
+  }
+
+  if (data.hasOwnProperty('reviveCmd') &&
+      typeof(data.reviveCmd) !== 'string') {
+    return undefined;
+  }
+
+  if (data.hasOwnProperty('reviveArgs') &&
+      typeof(data.reviveArgs) !== 'string') {
+    return undefined;
+  }
+
+  return data;
+};
+
+/**
+ * An internal interface representing the state we keep track of for each
+ * network node.
+ */
+interface IMonitoredNodeStatus {
+  address: string;
+  alive: boolean;
+  lastUpdate: Date;
+  lastResponseId: number;
+  networkId?: string;
+  reviveCmd?: string;
+  reviveArgs?: string[];
+};
+
+/**
+ * An internal interface representing the response of the ping command to the
+ * network node.
+ */
+interface INetVersionResponse {
+  id: number;
+  result: string;
+};
+
+/**
+ * The actual monitor implementation. It contains an express application and
+ * last known node status.
+ */
+class Monitor {
+  /**
+   * The interval at which the monitor pings the instances. If the last update
+   * time was longer than this interval, the instance is considered dead.
+   */
+  static k_HEARTBEAT_INTERVAL = 5000;
+
+  /**
+   * The web server
+   */
+  server: express.Application;
+
+  /**
+   * The current state of the nodes
+   */
+  nodesStatus: IMonitoredNodeStatus[];
+
+  /**
+   * The timer used to ping nodes
+   */
+  timer: NodeJS.Timer;
+
+  private validateNetVersionResponse(data: any) : INetVersionResponse | undefined {
```

I think in general if something is not valid, we should throw an error instead of return `undefined`.